src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -565,7 +571,7 @@ private function validateAndDenormalize(array $types, string $currentClass, stri
                 // PHP's json_decode automatically converts Numbers without a decimal part to integers.
                 // To circumvent this behavior, integers are converted to floats when denormalizing JSON based formats and when
                 // a float is expected.
-                if (Type::BUILTIN_TYPE_FLOAT === $builtinType && \is_int($data) && null !== $format && str_contains($format, JsonEncoder::FORMAT)) {
+                if (Type::BUILTIN_TYPE_FLOAT === $builtinType && \is_int($data) && null !== $format && str_contains($format, static::JSON_ENCODER_FORMAT)) {
```

Comment:
This is strange to me that we are using a constant there. Instead of introducing service "state", what about setting that value in the context (and updating the `AbstractObjectNormalizerContextBuilder` consequently)?

Then, we could set the `JsonEncoder::FORMAT` value in the `defaultContext` instead and do something like:
```php
$jsonFormat = $context[self::JSON_ENCODER_FORMAT] ?? $this->defaultContext[self::JSON_ENCODER_FORMAT] ?? JsonEncoder::FORMAT;
```